src/scripts/main.js

```diff
@@ -1,3 +1,191 @@
 'use strict';
 
-// write your code here
+const startButton = document.querySelector('.start');
+
+startButton.insertAdjacentHTML('afterend', `
+  <button class="button restart hidden">Reset</button>
+  `);
+
+const restartButton = document.querySelector('.restart');
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const fieldCells = document.querySelectorAll('.field-cell');
+const scoreSpan = document.querySelector('.game-score');
+
+const getRandomCellValue = () => {
+  return Math.floor(Math.random() * 10) < 9 ? 2 : 4;
+};
+
+const getFreeRandomCell = () => {
+  const freeCells = [...fieldCells].filter(el => el.classList.length === 1);
+  const randomIndex = Math.floor(Math.random() * freeCells.length);
+
+  return freeCells[randomIndex];
+};
+
+const fillNewCell = () => {
+  const cell = getFreeRandomCell();
+  const value = getRandomCellValue();
+
+  cell.textContent = value;
+  cell.classList.add(`field-cell--${value}`);
+};
+
+const toggleHidden = (...elements) => {
+  elements.forEach(element => {
+    element.classList.toggle('hidden');
+  }); ;
+};
+
+startButton.addEventListener('click', () => {
+  fillNewCell();
+  fillNewCell();
+  toggleHidden(startButton, restartButton, messageStart);
+  document.addEventListener('keydown', handleMove);
+});
+
+restartButton.addEventListener('click', () => {
+  [...fieldCells].map(item => {
```

Comment:
It's better to use `forEach` instead of `map` here, as you are not using the returned array and just performing a side effect 🤔